controller/internal/enforcer/nfqdatapath/afinetrawsocket/afinetrawsocket_windows.go

```diff
@@ -0,0 +1,66 @@
+// +build windows
+
+package afinetrawsocket
+
+import (
+	"errors"
+	"unsafe"
+
+	"go.aporeto.io/trireme-lib/controller/internal/windows/frontman"
+	"go.aporeto.io/trireme-lib/controller/pkg/packet"
+)
+
+type rawsocket struct {
+}
+
+type WindowsPacketMetadata struct {
+	PacketInfo frontman.PacketInfo
+	IgnoreFlow bool
+}
+
+const (
+	// RawSocketMark is the mark asserted on all packet sent out of this socket
+	RawSocketMark = 0x63
+	// NetworkRawSocketMark is the mark on packet egressing
+	//the raw socket coming in from network
+	NetworkRawSocketMark = 0x40000063
+	//ApplicationRawSocketMark is the mark on packet egressing
+	//the raw socket coming from application
+	ApplicationRawSocketMark = 0x40000062
+)
+
+// SocketWriter interface exposes an interface to write and close sockets
+type SocketWriter interface {
+	WriteSocket(buf []byte, version packet.IPver, data interface{}) error
+}
+
+// CreateSocket returns a handle to SocketWriter interface
+func CreateSocket(mark int, deviceName string) (SocketWriter, error) {
+	return &rawsocket{}, nil
+}
+
+func (sock *rawsocket) WriteSocket(buf []byte, version packet.IPver, data interface{}) error {
+	w, _ := data.(*WindowsPacketMetadata)
+	if w == nil {
```

Comment:
This needs a type check (and thus my problem with interfaces) 

w, ok := data.(*WindowsPacketMetadata)
if !ok {
...
}

Reply:
fixed in code